fix: remove unreachable else branch in from_unicode (#1345)
collective/icalendar: CHANGES.rst

Minor changes
~~~~~~~~~~~~~

> - Remove unreachable ``else`` branch in ``from_unicode`` and ``to_unicode`` in ``parser_tools.py``, raise ``TypeError`` for invalid input instead. :issue:`641`

Reviewer: With the release of v7.1.1, change log entries should be in a new file. See https://icalendar.readthedocs.io/en/stable/contribute/index.html#change-log-entry-format. Please remove this change, and put the following content into `/news/641.bugfix`.

```suggestion
Removed unreachable ``else`` branch from :func:`~icalendar.parser_tools.from_unicode` and :func:`~icalendar.parser_tools.to_unicode` in :file:`parser_tools.py`, raising a ``TypeError`` for invalid input instead. @shaktipy
```
